File: src/main/java/com/range/rpms/user/service/AuthService.java
package com.range.rpms.user.service;

import com.range.rpms.user.dto.UserLoginRequest;
import com.range.rpms.user.dto.UserRegisterRequest;
import org.springframework.stereotype.Service;

@Service
public interface AuthService {
    String login(UserLoginRequest userLoginRequest);
    String register (UserRegisterRequest userRegisterRequest);
}

File: src/main/kotlin/com/example/rpms/users/service/AuthService.kt
package com.example.rpms.users.service

import com.example.rpms.users.dto.AuthResponse
import com.example.rpms.users.dto.LoginRequest
import com.example.rpms.users.dto.RegisterRequest

interface AuthService {
    fun login(loginRequest: LoginRequest): AuthResponse
    fun register(registerRequest: RegisterRequest)
    fun resetPassword(token: String)
    fun forgotPassword(email: String)
    fun twoFactoryAuthRequest(email:String)
    fun acceptTwoFactoryAuthRequest(token: String):AuthResponse
    fun logout()
}